### PR TITLE
fix #2811 by updating RID synthesis

### LINKED ISSUE #2811: DNX RID synthesis is broken
After @moozzyk 's change, `IRuntimeEnvironment.OperatingSystemVersion` provides the Windows Product Version numbers (i.e. 8, 8.1, etc.) instead of the actual underlying OS versions (6.1, 6.2, etc.). This breaks our RID synthesis code.

I'll have a fix shortly, this is just to track the work.

/cc @BrennanConroy @muratg @moozzyk @davidfowl

### PR DESCRIPTION
**Pull request by anurse**
Fixes #2811 

I'm not sure there's a reliable way to test this unfortunately...

/cc @muratg @davidfowl @moozzyk @BrennanConroy

### CODE REVIEW

**Comment by BrennanConroy** [comment]
:shipit: but let Pawel look first

**Comment by moozzyk** [comment]
:ship::it:

**Comment by anurse** [comment]
You guys beat `build verify` so I'll wait for that and Travis before merging :)

File: src/Microsoft.Dnx.Runtime/ApplicationHostContext.cs [hunk 1]
```diff
@@ -182,6 +182,11 @@ private static Result InitializeCore(ApplicationHostContext context)
             if (lockFileLookup != null)
             {
                 var target = SelectTarget(context, context.LockFile);
+                if (Logger.IsEnabled && string.IsNullOrEmpty(target.RuntimeIdentifier))
+                {
+                    // REVIEW(anurse): Is there ever a reason we want to use the RID-less target now?
```
Comment (anurse) [inline src/Microsoft.Dnx.Runtime/ApplicationHostContext.cs]: @davidfowl see this question. Thoughts?

File: src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs [hunk 2]
```diff
@@ -46,23 +46,22 @@ public static IEnumerable<string> GetAllRuntimeIdentifiers(this IRuntimeEnvironm
             else
             {
                 var arch = env.RuntimeArchitecture.ToLowerInvariant();
-                var parsedVersion = Version.Parse(env.OperatingSystemVersion);
-                if (parsedVersion.Major == 6 && parsedVersion.Minor == 1)
+                if(env.OperatingSystemVersion.Equals("7.0", StringComparison.Ordinal))
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs]: spaces

**Comment by moozzyk** [comment]
I actually think tests need to be updated...

**Comment by anurse** [comment]
@moozzyk Should have them all updated now.

**Comment by moozzyk** [comment]
:+1:

**Pull request closed without merging**